File: c/04-control-flow/loops.c
#include <stdio.h>

int main(void) {
    // for loop: sum the integers 1 through 5
    int sum = 0;
    for (int i = 1; i <= 5; i++) {
        sum += i;
    }
    printf("Sum 1..5 = %d\n", sum);

    // while loop: count down from 3
    int count = 3;
    while (count > 0) {
        printf("Countdown: %d\n", count);
        count--;
    }

    // do-while loop: runs the body before testing the condition
    int attempts = 0;
    do {
        attempts++;
        printf("Attempt %d\n", attempts);
    } while (attempts < 2);

    return 0;
}

File: java/04-control-flow/Loops.java
public class Loops {
    public static void main(String[] args) {
        // Classic for loop with an index counter
        System.out.print("Squares:");
        for (int i = 1; i <= 5; i++) {
            System.out.print(" " + (i * i));
        }
        System.out.println();

        // Enhanced for loop (for-each) over an array
        String[] languages = {"Java", "Kotlin", "Scala"};
        for (String lang : languages) {
            System.out.println("JVM language: " + lang);
        }

        // while loop: condition checked before each iteration
        int countdown = 3;
        while (countdown > 0) {
            System.out.println("T-minus " + countdown);
            countdown--;
        }
        System.out.println("Liftoff!");

        // do-while loop: body runs at least once
        int attempts = 0;
        do {
            attempts++;
            System.out.println("Attempt " + attempts);
        } while (attempts < 2);
    }
}
